tests/unit/lib/build_module/test_build_strategy.py

```diff
@@ -193,6 +193,30 @@ def test_build_layers_and_functions(self, mock_copy_tree, mock_path):
             str(mock_path(given_build_dir, self.function1_2.name)),
         )
 
+    def test_build_single_function_definition_image_functions_with_same(self, mock_copy_tree, mock_path):
```

Comment:
should we also add an integ test for this?

Reply:
sure